modules/react/common/README.md

```diff
@@ -237,3 +240,130 @@ Theming documentation has its own README. You can find it [here](./lib/theming/R
 
 Bidirectionality is provided by Theming. You can find Theming documentation
 [here](./lib/theming/README.md#bidirectionality).
+
+## Component Functions
+
+### `createComponent`
+
+`createComponent` is a convenience function that makes it easier to create components that
+[forward refs](https://reactjs.org/docs/forwarding-refs.html) and take an `as` prop. The `as` prop
+allows you to change the default element rendered by the component.
+
+```tsx
+interface MyComponentProps {
+  prop1: string;
+  prop2: number;
+  children: React.ReactNode;
+}
+
+export const MyComponent = createComponent('div')({
+  displayName: 'MyComponent', // the component's name in the React dev tools
+  Component: (({ prop1, prop2, children, ...elemProps }: MyComponentProps), ref, as) => {
+    return <Box as={as} {...elemProps}>{children}</Box>
+  })
+})
+```
+
+Using a component created via `createComponent`:
+
+```tsx
+<MyComponent ref={ref} as="aside">
+  Content
+</MyComponent>
+```
+
+`MyComponent` is returning a type that looks like `ElementComponent<'div', MyComponentProps>`. The
+`'div'` tells you what the default element is of the element component.
+
+Notice that `MyComponentProps` does not extend `React.HTMLAttributes<HTMLDivElement>`. The `as` prop
+changes the interface of the component to match that of the `aside` element which is a
+`React.HTMLAttributes<HTMLElement>`. For this reason, the interface cannot be known statically ahead
+of time. Components using `createComponent` are starting to not export the prop interface. For this
+reason, [ExtractProps](#extractprops) was created.
+
+The first parameter of `createComponent` might not be defined. If this is the case, the type
+returned is a `Component` instead of `ElementComponent`. `Component`s do not have any HTML
+attributes associated with them. This usually means the component doesn't return a real element.
+This is used for compound components that do not require an element for accessibility reasons and
+only provide a common model for the subcomponents.
+
+### `ExtractProps`
+
+`ExtractProps` can be used in place of importing a prop interface directly. If you wish to extend a
+Canvas Kit component by extending it's props, use `ExtractProps`.
+
+```tsx
+interface MyNewComponentProps extends ExtractProps<typeof MyComponent> {
```

Comment:
i would write this as `interface MyNewComponentProps extends ExtractProps<typeof Card> {`

so it's clear that it's a canvas kit component. super nit

Reply:
`MyComponent` is referring to the component defined further up the doc on line 259.